abstract/encoding.go

```diff
@@ -170,13 +193,31 @@ func (de *decoder) value(v reflect.Value, depth int) error {
 			}
 		}
 
-	case reflect.Slice, reflect.Array:
+	case reflect.Slice:
+		if v.IsNil() {
+			panic("slices must be initialized to correct length before decoding")
+		}
+		fallthrough
+	case reflect.Array:
 		l := v.Len()
 		for i := 0; i < l; i++ {
 			if err = de.value(v.Index(i), depth+1); err != nil {
 				return err
 			}
 		}
+
+	case reflect.Int:
```

Comment:
I'm afraid hunting down a bug in a client (or a server, when it has to receive) will be a REAL headache - a real heisenbug. Imagine your app working fine in test development, and when you deploy it, all of a sudden the thing crashes on the clients and you have no log?

But then I don't know how many 'int's are currently in a structure that is being (Un)Marshalled by the Suite.Write, which will also be difficult to change.

Can't a basic crypto-suite make some enforcements about what type of int to use, if it's a possible problem?